mobile-ui/components/Fan.jsx

```diff
@@ -1,26 +1,22 @@
 import { Text, View, ViewBase, Alert, Image } from "react-native";
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import ToggleSwitch from "toggle-switch-react-native";
 import Slider from "@react-native-community/slider";
 
-function Fan({ powerState }) {
-    const [isOn, setIsOn] = useState(powerState);
-    const toggleState = () => {
-        if (isOn == true) {
-            setpower(0);
-        }
-        setIsOn((previousState) => !previousState);
-
-        // alert("Toggled")
-    };
-    const [power, setpower] = useState(0);
+function Fan({ powerState, callback }) {
+    const [power, setPower] = useState(powerState)
+    const handleToggle = () => {
+        callback(powerState == 0 ? 2 : 0);
+    }
+    
+    useEffect(() => {setPower(powerState)}, [powerState])
```

Comment:
I think we can remove the power state, because it's just an instance of powerState prop. If you have trouble with rerender, it may be caused by rendering conditon (use key to resovle)